lib/valkyrie/controller_concerns/model_controller_behavior.rb

```diff
@@ -17,9 +17,16 @@ def create
       authorize! :create, @form.model
       if @form.validate(model_params)
         @form.sync
-        obj = nil
+        obj = @form.model
         persister.buffer_into_index do |persist|
-          obj = persist.save(model: @form)
+          appender = FileAppender.new(storage_adapter: Valkyrie.config.storage_adapter, persister: persist, files: @form.try(:files) || [])
```

Comment:
It's not part of the default interface - PageForm doesn't have one. We could make it part of the default?